### ISSUE TITLE
[googlecloudmonitoring] config collection interval & more flexible metric filtering

### ISSUE THREAD

**Issue by chenlujjj** [open-telemetry/opentelemetry-collector-contrib#36898]
### Component(s)

receiver/googlecloudmonitoring

### Is your feature request related to a problem? Please describe.

1. The default collection interval is 5 mins, which is too large for monitoring
2. The current metric filtering is not flexible enough, if we want to collect a bunch of metrics, the list would be too long

### Describe the solution you'd like

1. decrease the default collection interval
2. enrich the metric filtering capabilities, such as include/exclude, regex, etc

**Comment by dashpole** [comment]
A 5 minute interval was requested by the cloud monitoring team, so I don't think we should lower the default.  Lower intervals can cause query performance problems.

Better filtering makes sense.  We should try and be consistent with other filtering-related configuration in collector components if we can, rather than invent something new here.

**Comment by chenlujjj** [comment]
Hi @dashpole
> A 5 minute interval was requested by the cloud monitoring team,
could you tell me where I can find the request? 5 minute interval means that the alert can delay at most 5 minutes, and I'm worried about that.

**Comment by dashpole** [comment]
I'm a Googler, so I reached out to them internally.  @kjordy are there any details you can share here as to why a 5 minute interval is recommended for scraping ListTimeSeries?

**Comment by chenlujjj** [comment]
Is it possible to use a smaller interval restriction, like 1 minute? 

I also checked the [documentation](https://docs.splunk.com/observability/en/gdi/get-data-in/connect/gcp/gcp-connect.html#connect-to-splunk-observability-cloud-and-start-the-integration) of Splunk SignalFx connection to GCP, the minimum poll rate can be 1 minute.

<img width="922" alt="image" src="https://github.com/user-attachments/assets/681d5a94-e065-4ac4-9902-89bb0e0281c2" />

**Comment by chenlujjj** [comment]
hello ,  @kjordy, could you give extra information of the collection interval ?

**Comment by dashpole** [comment]
This is the response I got from the monitoring API team:

This is a balance between freshness and QPS. This is also the default of many 3P vendors like [Datadog](https://docs.datadoghq.com/developers/guide/data-collection-resolution-retention/). There are both [costs](https://cloud.google.com/stackdriver/pricing) and [quotas](https://cloud.google.com/monitoring/quotas#api_quotas) associated with queries. Scraping at a low interval may be ok for a small number of metrics but is not recommended for doing bulk export.

**Comment by chenlujjj** [comment]
Hi @dashpole , thanks for the clarification, I understand the default interval setting now.

I think maybe it's better to separate the default interval and the minimum interval settings ?

Datadog has 5 min as default and 1 min as minimum:

<img width="967" alt="Image" src="https://github.com/user-attachments/assets/84960c64-9d5b-4709-b6eb-e1b9f11da31f" />

**Comment by dashpole** [comment]
yeah, SGTM.  Lets set a minimum of 1 minute.  We should also document the costs of setting a lower interval.

**Comment by chenlujjj** [comment]
ok, I'll make a PR for it.

**Comment by chenlujjj** [comment]
> We should try and be consistent with other filtering-related configuration in collector components if we can, rather than invent something new here.

@dashpole Regarding the " filtering-related configuration ", can you recommend one component that I can refer to implement this feature ?

**Comment by dashpole** [comment]
Ah, I didn't realize there was already a filter config in the monitoring API.  We should just use our own configuration in that case.
